### PR TITLE
ApprovalRequest Classification Migration

### PR DESCRIPTION
**Pull request by tharmaman**
Migrated ApprovalRequest.ClassificationEnum to ApprovalRequest.Classification.

- Created an **ApprovalRequestClassificationDAO** and **ApprovalRequestClassificationModel**
- Added a jrl for approvalRequestClassifications which should only exist on foam
- Added a translation for ApprovalRequest.Classification and for some locales that exist foam side
- Reworked various logic to use an ApprovalRequestClassification id instead of an ApprovalRequestClassificationEnum
-  Replaced classificationEnum on ApprovalRequest tableColumns && searchColumns to instead use classification
- Made ApprovalRequest.classification.tableCellFormatter use `foam.u2.view.ReferenceToSummaryCellFormatter`
- Used the translationService in **ApprovalRequestClassification.toSummary()**

### CODE REVIEW

File: src/services [hunk 1]
```diff
@@ -1173,3 +1173,23 @@ p({
...
 })
+
+p({
+  "class":"foam.nanos.boot.NSpec",
+  "name":"approvalRequestClassificationDAO",
+  "serve":true,
+  "serviceScript": """
+    return new foam.dao.EasyDAO.Builder(x)
+      .setGuid(true)
+      .setAuthorize(false)
```
Comment (zahnno) [inline src/services]: Sure we don't want this authorized? I think we do.

**Review by zahnno — COMMENTED** [review]
Good to see this - always thought the approvalRequestEnum was odd.

File: src/foam/nanos/approval/ApprovalRequest.js [hunk 2]
```diff
@@ -723,12 +699,13 @@ foam.CLASS({
...
+        return this.classification.id;
       },
       javaCode: `
-        return getClassificationEnum().getLabel();
+        return getClassification();
       `
     },
+    // TODO: remove this when we remove classificationEnum
     {
       name: 'getClassificationEnumIsSet_',
```
Comment (minsunny) [inline src/foam/nanos/approval/ApprovalRequest.js:710]: Do we still need this too?

File: src/foam/nanos/approval/ApprovalRequestClassification.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    {
+      name: 'toSummary',
+      type: 'String',
+      code: function() {
+        return this.translationService.getTranslation(foam.locale, this.id + ".name", this.name);
+      },
+      javaCode: `
+        TranslationService ts = (TranslationService) getX().get("translationService");
+        Subject subject = (Subject) getX().get("subject");
+        String locale = ((User) subject.getRealUser()).getLanguage().getCode().toString();
```
Comment (jlhughes) [inline src/foam/nanos/approval/ApprovalRequestClassification.js]:
```suggestion
        String locale = ((User) subject.getRealUser()).getLanguage().getCode().toString();
```
String locale = (String) getX().get("foam.nanos.auth.LocaleSupport.CONTEXT_KEY");

**Review by nanoscott — APPROVED** [review]

**Comment by kgrgreer** [comment]
No comments other than those raised by other reviewers.

File: src/services (hunk 1 above)
Comment (tharmaman) [reply src/services]: Removed and granted approvalrequestclassification.read.* to basicUser

File: src/foam/nanos/approval/ApprovalRequestClassification.js (hunk 3 above)
Comment (tharmaman) [reply src/foam/nanos/approval/ApprovalRequestClassification.js]: Done.

File: src/services [hunk 4]
```diff
@@ -1168,3 +1168,22 @@ p({
...
   "serviceClass": "foam.nanos.medusa.MedusaStatusWatcher"
 })
+
+p({
+  "class":"foam.nanos.boot.NSpec",
+  "name":"approvalRequestClassificationDAO",
+  "serve":true,
+  "serviceScript": """
+    return new foam.dao.EasyDAO.Builder(x)
+      .setGuid(true)
```
Comment (jlhughes) [inline src/services]:
```suggestion
      .setFuid(true)
```
Use our new Fuid on anything new.  We'll slowly switch everything over.

**Review by zahnno — APPROVED** [review]

**Pull request merged**